### PR TITLE
[Merged by Bors] - Create separate connection to state DB for API

### LINKED ISSUE #5583: setup a separate sqlite connection pool for consensus and api
sqlite connection is taken from the pool for the duration of the database request, e.g if we read a collection it will take a connection for the whole duration.

as of now API is not very rich and mostly used for feeding data into explorer, but with addition of new methods node may experience connection pool saturation. it may lead to missing critical timings in the consensus, and node will lag in processing layers, and possibly will have to download more data from peers.

implementation wise it makes sense to create another *Database instance but with a different pool, and pass it to the API.
note that API should not interact with database through various wrappers hidden with interfaces, how it was implemented before.

### PR DESCRIPTION
**Pull request by kacpersaw**
## Description

This pull request introduces a separate SQL connection pool for the API interface. This change aims to isolate the API's database interactions from the main application pool to prevent exhausting all connections.

Closes #5583

### CODE REVIEW

**Review by ivan4th — APPROVED** [review]

File: node/node.go [hunk 1]
```diff
@@ -1980,6 +1987,19 @@ func (app *App) setupDBs(ctx context.Context, lg log.Log) error {
...
 	}
 	app.db = sqlDB
+
+	apiDBLog := app.addLogger(ApiStateDBLogger, lg).Zap()
+	apiSqlDB, err := statesql.Open("file:"+filepath.Join(dbPath, dbFile), []sql.Opt{
+		sql.WithReadOnly(),
+		sql.WithLogger(apiDBLog),
+		sql.WithConnections(app.Config.API.DatabaseConnections),
+		sql.WithMigrationsDisabled(),
+	}...)
```
Comment (fasmat) [inline node/node.go]: NIT: the slice definition isn't needed 🙂 

```suggestion
	apiSqlDB, err := statesql.Open("file:"+filepath.Join(dbPath, dbFile),
		sql.WithReadOnly(),
		sql.WithLogger(apiDBLog),
		sql.WithConnections(app.Config.API.DatabaseConnections),
		sql.WithMigrationsDisabled(),
	)
```

**Review by fasmat — COMMENTED** [review]
Is this safe to do? I know the API opens the DB readonly, but the node changes the data in the DB.

What happens if data is changed by the node while the API reads from the DB? Like inserting ATXs while API queries those ATXs?

**Comment by ivan4th** [comment]
> What happens if data is changed by the node while the API reads from the DB? Like inserting ATXs while API queries those ATXs?

If there's a long-running `SELECT` than write transaction will not begin until it ends
There can be situation where writes can happen between several reads but given the writes that may happen are "append-only" I don't immediately see a possibility for inconsistencies.
If there's such a possibility we need to use read transactions, but this is not supported by `sql/database` currently, as all the transactions started are actually immediate (they acquire write lock immediately). Note that long-running read transactions will block writes (causing `SQLITE_BUSY`), though.

**EDIT:** I made this mistake again in this comment, sorry :)
We use WAL and read transactions do no block writes. Read transactions only block WAL checkpointing.
But there can be only one active write transaction at a given moment.

**Review by fasmat — APPROVED** [review]

**Comment by ivan4th** [comment]
Updated my comment above b/c I made a mistake in it regarding db locking

**Comment by ivan4th** [comment]
(but this doesn't block this PR, it's ok to merge)

**Comment by kacpersaw** [comment]
bors merge

**Pull request closed without merging**